src/ValidatingFunctionWrapper.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\CloudFunctions;
+
+use LogicException;
+use ReflectionFunctionAbstract;
+
+abstract class ValidatingFunctionWrapper extends FunctionWrapper
```

Comment:
I don't like the idea of creating a `ValidatingFunctionWrapper` class. I don't think the distinction between the different wrappers justifies this level of complexity.

What if you renamed `ValidatingFunctionWrapper` to `FunctionValidationTrait`, make it a `trait`, and have it implement the same methods as you already have here (except for `__construct`, which you could leave in `FunctionWrapper` and just make `validateFunctionSignature` protected). Then implement this trait in `HttpFunctionWrapper` and `CloudEventFunctionWrapper`, and this'll be good to go.

Reply:
Done, I think that's an improvement.